src/nn.rs

```diff
@@ -8,18 +8,16 @@ use utils::samples_output_to_matrix;
 
 /// Represents a Neural Network with layers, inputs and outputs
 pub struct NeuralNetwork<T: Activation> {
-    activation: T,
-    layers: Vec<NeuralLayer>,
+    layers: Vec<NeuralLayer<T> >,
     samples: Vec<Sample>,
     error_fn: Option<Box<Fn(f64)>>
 }
 
-impl<T: Activation> NeuralNetwork<T> {
+impl <T: Activation> NeuralNetwork<T> {
```

Comment:
I'm pretty sure we need it since the network is predicated on using a generic activation of which the layers must implement. maybe? We can try it and see if the compiler loses it.